src/ast/ast_validations.c

```diff
@@ -1573,112 +1582,90 @@ static AST_Validation _ValidateScopes
 	return !ErrorCtx_EncounteredError() ? AST_VALID : AST_INVALID;
 }
 
-static VISITOR_STRATEGY _default_visit
-(
-	const cypher_astnode_t *n,
-	bool start,
-	ast_visitor *visitor
-) {
-	ASSERT(n != NULL);
-
-	return VISITOR_RECURSE;
-}
-
-// void AST_MappingInit() {
-// build the global mapping from ast-node-type to visiting function
-bool AST_ValidationsMappingInit() {
+// build the global mapping from ast-node-type to visiting functions
+bool AST_ValidationsMappingInit(void) {
 	// create a mapping for the validations
-	validations_mapping = AST_Visitor_mapping_new();
 
-	// initialize all entires with the default visit function
-	for(uint i = 0; i < 256; i++) {
-		AST_Visitor_mapping_register(validations_mapping, i, _default_visit);
+	// set default entries
+	for(uint i = 0; i < 114; i++) {
+		validations_mapping[i] = _default_visit;
 	}
```

Comment:
see if this loop can be removed if we statically init `validations_mapping` to `{_default_visit}`

Reply:
Unfortunately, seems that we can't.